src/test/java/uk/gov/ch/oauth/session/SessionFactoryTest.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ch.oauth.session;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionImpl;
+import uk.gov.companieshouse.session.store.Store;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class SessionFactoryTest {
+
+    @Mock
+    private Store store;
+
+    @InjectMocks
+    SessionFactory sessionFactory;
+
+    @Spy
+    Session oldSession = new SessionImpl();
+
+    @Test
+    public void testRegenerateSession() {
+        oldSession.setCookieId("OldSessionID" + "signature");
```

Comment:
Please make this a `static final` constant value and compare against that later.